vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -84,6 +102,47 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
         Mockito.doThrow(EmptyResultDataAccessException.class).when(textBlockRepo).deleteById(textBlockId);
         managerToTest.deleteTextBlockById(textBlockId, "slideId_1");
     }
+    
+    @Test(expected = BlockDoesNotExistException.class)
+    public void test_deleteSpaceBlockById_forNonExistentId() throws BlockDoesNotExistException {
+        String spaceBlockId = "notARealId";
+        when(contentBlockRepository.findById("notARealId")).thenReturn(Optional.empty());
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(spaceBlockRepo).deleteById(spaceBlockId);
+        managerToTest.deleteSpaceBlockById(spaceBlockId,"slideId_1");
+    }
+    
+    @Test
+    public void test_deleteSpaceBlockById_valid() throws BlockDoesNotExistException {
+        SpaceBlock spaceBlock = new SpaceBlock();
+        managerToTest.saveSpaceBlock(spaceBlock);
+        
+        Mockito.verify(spaceBlockRepo).save(spaceBlock);
+        
+    }
+    
+    @Test
+    public void test_editSpaceBlock_valid() throws BlockDoesNotExistException {
+        String spaceBlockId = "realId";
+        when(contentBlockRepository.findById(spaceBlockId)).thenReturn(Optional.of(contentBlock));
+        managerToTest.deleteSpaceBlockById(spaceBlockId,"slideId_1");
+        Mockito.verify(spaceBlockRepo).deleteById(spaceBlockId);
+        
+    }
+    
+    @Test
+    public void test_addSpaceBlock_success() {
+        String slideId = "slideId";
+        ISlide slide = null;
+        String title = "this is a space block";
+        Integer contentOrder = 2;
+        Space space = new Space();
+        ISpaceBlock spaceBlock = new SpaceBlock();
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(spaceBlockFactory.createSpaceBlock(slide, title, (ISpace)space)).thenReturn(spaceBlock);
+        when(spaceBlockRepo.save((SpaceBlock)spaceBlock)).thenReturn((SpaceBlock) spaceBlock);
+        ISpaceBlock createdBlock = managerToTest.createSpaceBlock(slideId, title, contentOrder, space);
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
+    }
 
```

Comment:
what about the `getSpaceBlock` method?

Reply:
?

Reply:
I am writing test cases and increased the code coverage from 38% to 72%